architecture/verifier/freshness.md

```diff
@@ -0,0 +1,101 @@
+## Freshness
+
+In order to prevent replay attacks, Veraison employs a two-stage
+challenge-response protocol for verification. First, a verification session is
+created, and a nonce for that session is established. The nonce is used as a
+challenge when generating the attestation report/token which is presented in
+the second stage that results in a signed attestation result produced by the
+verifier.
+
+Since the way the nonce is incorporated in the attestation token depends on
+the specifics of the attestation scheme, the core Veraison verifier only
+propagates the session nonce (established in the first stage), along with the
+evidence extracted from the attestation token. There are three options for
+how/where token freshness can be validated:
+
+1. by the attestation scheme's Evidence Handler plugin
+2. by the deployment-specific policy associated with the scheme
+3. left to the Relying Party verifier based on the attestation result (which
+   includes both the session nonce and the extracted evidence)
+
+The following diagram shows the (simplified) verification flow,
+illustrating how the nonce travels through it and how it may be used to ensure
+freshness.
+
+![nonce flow diagram](nonce-sequence.png)
+
+1. Verification is initiated by the client (the Attester or the Relying Party,
+   depending on the [interaction
+   model](https://www.rfc-editor.org/rfc/rfc9334.html#name-topological-patterns)
+   used) with a new session request. The client can optionally specify the
+   nonce to be used for this verification session, or the size of the nonce to
+   be generated by Veraison. Either way, the nonce is limited to be between 8
+   and 64 bytes long (see [EAT `eat_nonce`
+   claim](https://www.ietf.org/archive/id/draft-ietf-rats-eat-21.html#name-eat_nonce-eat-nonce-claim)).
+2. The nonce (whether provided or generated) is associated with the created
+   session, and is returned in the response to the new session request.
+3. (The assumption is that) the returned nonce is then used as the challenge to
+   the attester when generating the attestation token.
+4. The attestation token/evidence (that embeds the nonce/challenge) is then
+   submitted to the session URL for verification.
+5. The verification frontend retrieves the nonce it has associated with the
+   session and attaches it to the received attestation token, and sends to the
+   verifier on the VTS backend via a `IVTSClient.GetAttestation()` call.
```

Comment:
```suggestion
   VTS backend via a `IVTSClient.GetAttestation()` call.
```

Reply:
I'm specifically using "verifer on the VTS" rather than "VTS" to correpond to the diagram, where VTS consists of the verifier (the generic pipeline code), the scheme handler (plugin), and the policy agent.

Reply:
Sure that is fine!